### PR TITLE
Fix cross signing setup from settings screen

### PR DESCRIPTION
**Pull request by dbkr**
We previously called bootstrap without the cache callbacks set up
which meant it prompted you for you password once for each of the
three keys it fished out of SSSS. Instead, display the same dialog
we display from the verify toast.

Also, in the case where you hadn't verified yet, the button didn't
do anything at all, so make it trigger a verification & change the
caption appropriately.

Add a state to reflect when cross signing is set up but SSSS isn't
(ie. user hasn't yet set up backup / set a passphrase).

Fixes https://github.com/vector-im/element-web/issues/17761
Type: defect

Screenshot 2021-08-18 at 18 53 53
Screenshot 2021-08-18 at 18 54 01
Screenshot 2021-08-18 at 19 21 39
Screenshot 2021-08-18 at 19 07 58

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix cross signing setup from settings screen ([\#6633](https://github.com/matrix-org/matrix-react-sdk/pull/6633)). Fixes vector-im/element-web#17761 and vector-im/element-web#17761.

Preview: https://611d51d21916c900ed4a846e--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/settings/CrossSigningPanel.tsx [hunk 1]
```diff
@@ -176,10 +187,14 @@ export default class CrossSigningPanel extends React.PureComponent<{}, IState> {
             summarisedStatus = <p>{ _t(
                 "Your homeserver does not support cross-signing.",
             ) }</p>;
-        } else if (crossSigningReady) {
+        } else if (crossSigningReady && crossSigningPrivateKeysInStorage) {
             summarisedStatus = <p>✅ { _t(
                 "Cross-signing is ready for use.",
             ) }</p>;
+        } else if (crossSigningReady && !crossSigningPrivateKeysInStorage) {
+            summarisedStatus = <p>⚠️ { _t(
```
Comment (turt2live) [inline src/components/views/settings/CrossSigningPanel.tsx:195]: emoji should be in the string for RTL language translation, unless we consider this a design constraint.

**Review by turt2live — APPROVED** [review]
it seems sensible, though all of the SSSS/Cross-Signing magic callbacks terrify me.

**Comment by amshakal** [comment]
This looks fine from the design end as it makes the flows more consistent. A quick question: is there a reason why some buttons are left aligned and some right aligned?

**Comment by dbkr** [comment]
The ones in the dialogs are right-aligned but the the left-aligned ones are just in the settings panel.

**Comment by amshakal** [comment]
Ah of course, sorry, I got a little confused ^_^

**Pull request merged**